test/models/ExtendsOrder.mo

```diff
@@ -21,14 +23,25 @@ package P
     model B
       model BT
         Real x;
-        parameter Real m = 0;
+        constant Real bla = 100;
+        // Are there references in which the scope remains BT, not bt?
+        parameter Real m(nominal=BT.bla, min=P.min_m, max=2 * max_m) = 0;
+        parameter Real n(nominal=bla, min=P.min_m, max=2 * max_m) = 0;
       equation
-        x = m;
+        x = m * max_m;
       end BT;
-      BT bt;
+      BT bt(bla=150);
     end B;
 
     extends A(redeclare model AT = BT); // Here we expect to find P.M.B.BT, not P.BT
```

Comment:
I think the way to do this is to shift all extends modifiers first to all "partially instantiated" classes, and only then extend. Because at least OMC applies the modifier for BT(bla=200) also on the BT in `extends A(redeclare model AT=BT)` does